src/transformers/models/character_bert/tokenization_character_bert.py

```diff
@@ -0,0 +1,979 @@
+            max_word_length=max_word_length,
+        )
+
+        # Pads a sequence of TOKENS up to a desired length
+        # NOTE: since this is zero, actual characters' ids are assigned
+        # the range 1-256 instead of 0-255. This means that all character
+        # ids are eventually shifted by a +1 so that 0 can be kept for padding.
+        self.pad_character_ids = [0] * max_word_length
+
+    @property
+    def do_lower_case(self):
+        return self.basic_tokenizer.do_lower_case
+
+    @property
+    def vocab_size(self):
+        return len(self.vocab)
+
+    @property
+    def mlm_vocab_size(self):
+        return len(self.mlm_vocab)
+
+    def get_vocab(self):
+        return dict(self.vocab, **self.added_tokens_encoder)
+
+    def get_mlm_vocab(self):
+        return self.mlm_vocab
+
+    def _tokenize(self, text, split_special_tokens=False):
+        split_tokens = []
+        if self.do_basic_tokenize:
+            for token in self.basic_tokenizer.tokenize(
+                text, never_split=self.all_special_tokens if not split_special_tokens else None
+            ):
+                split_tokens.append(token)
+        else:
+            split_tokens = whitespace_tokenize(text)
+        return split_tokens
+
+    def convert_mlm_token_to_id(self, token):
+        """Converts a token (str) into an id using the MLM vocab."""
+        return self.mlm_vocab.get(token, self.mlm_vocab.get(self.unk_token))
+
+    def convert_mlm_id_to_token(self, index):
+        """Converts an index (integer) into a token (str) using the MLM vocab."""
+        return self.mlm_ids_to_tokens.get(index, self.unk_token)
+
+    # NOTE: the following two methods have misleading names since we are
+    # working with character id lists instead of standard token ids.
+    # Changing these names breaks a lot of things so we keep them for now.
+    def _convert_token_to_id(self, token):
+        """Converts a token (str) into a list of character ids (integer)."""
+        if token == self.cls_token:
+            character_ids = self.cls_character_ids
+        elif token == self.sep_token:
+            character_ids = self.sep_character_ids
+        elif token == self.mask_token:
+            character_ids = self.mask_character_ids
+        elif token == self.pad_token:
+            character_ids = self.pad_character_ids
+        else:
+            token_bytes = token.encode("utf-8", "ignore")[: (self.max_word_length - 2)]
+            character_ids = [self.pad_character_id] * self.max_word_length
+            character_ids[0] = self.bow_character_id
+            for k, index in enumerate(token_bytes, start=1):
+                character_ids[k] = index
+            character_ids[len(token_bytes) + 1] = self.eow_character_id
+            # NOTE: we shift everything so that the PAD token
+            #       can be assigned the all zeros vector
+            character_ids = [(index + 1) for index in character_ids]
+        return character_ids
```

Comment:
if all the special tokens are correctly added, then this should only handle tokens that are not special.

Reply:
I'll try and figure out how to set up the special tokens properly -- hopefully this won't require too many overloads due to the differences in inputs between BERT (2d) / CharacterBERT (3d)

Reply:
@ArthurZucker I just pushed an update for this, but it required overwriting some methods such as _decode to make it work. Please let me know what you think about this. I can't see any other way to make this work 😞

Reply:
No worries if that is the only way! I usually strive for simplicity but I did not really understand the subtility of the character bert model! My bad 🤗